sourcetracker/_sourcetracker.py

```diff
@@ -9,80 +9,92 @@
+
+    Summary
+    -------
+    Checks if data contains `nan` or `null` values, and returns data as
+    type `np.int32`. If both `sources` and `sinks` are passed, columns must
+    match exactly (including order).
 
     Parameters
     ----------
-    feature_table : pd.DataFrame
-        Feature table where rows are features and columns are samples.
-    apply_fractional_value_correction : boolean
-        Fractional counts cause errors in indexing. Uses np.ceil to remove
-        them if True.
+    sources : pd.DataFrame
+        A dataframe containing count data. Must be castable to `np.int32`.
+    sinks : optional, pd.DataFrame or None
+        If not `None` a dataframe containing count data that is castable to
+        `np.int32`.
 
     Returns
     -------
-    feature_table : pd.DataFrame
-        Corrected feature table if changes have been made or original table if
-        no changes occurred. Will automatically be cast as dtype=np.int64
+    pd.Dataframe(s)
 
     Raises
     ------
     ValueError
-        If data contains values that Gibb's sampling cannot handle.
+        If `nan` or `null` values found in inputs.
+    ValueError
+        If any values are smaller than 0.
+    ValueError
+        If any columns of an input dataframe are non-numeric.
+    ValueError
+        If `sources` and `sinks` passed and columns are not identical.
     '''
-    if not np.isreal(feature_table.dtypes).all():
-        raise TypeError('Feature table contains one or more columns which are '
-                        'not numeric type. This is likely due to boolean, '
-                        'string, or other data types being present. Data must '
-                        'contain exclusively real-valued integers or floats.')
-
-    if np.isnan(feature_table.values).any():
-        raise ValueError('One or more values in the feature table is a `nan` '
-                         'or `null` value. Data must contain exclusively real-'
-                         'valued integers or floats.')
-
-    if not (np.ceil(feature_table.values) == feature_table.values).all():
-        if apply_fractional_value_correction is True:
-            return pd.DataFrame(np.ceil(feature_table.values).astype(np.int64),
-                                index=feature_table.index,
-                                columns=feature_table.columns)
-        else:
-            raise ValueError('Non-integer data in the feature table is not '
-                             'being corrected by `check_and_correct_data`. '
-                             'Data leaving this function must be np.int64 '
-                             'type. Either correct the table or pass the '
-                             '`apply_fractional_value_correction` to '
-                             '`check_and_correct_data`.')
-    return feature_table.astype(np.int64)
+    if sinks is not None:
+        dfs = [sources, sinks]
+    else:
+        dfs = [sources]
+
+    for df in dfs:
```

Comment:
It might ultimately be nice to make the error messages more informative here. For example, we could have them say which is the offending dataframe, what the offending samples are, etc. Could you create an issue for this? I'd be happy to take it, but we don't need to do it for this release. If we don't have that information in the error messages, we're going to be fielding questions to help users find it.